pytest_plugins/rerun_rp/rerun_rp.py

```diff
@@ -100,58 +58,108 @@ def pytest_addoption(parser):
+    parser.addoption("--sat-version", help=help_text)
 
 
 @pytest.hookimpl(tryfirst=True)
 def pytest_collection_modifyitems(items, config):
     """
-    Collects and modifies tests collection based on pytest options to select tests marked as
-    failed/skipped and user specific tests in Report Portal
+    Collects and modifies test collection based on the pytest options to select the tests marked as
+    failed/skipped and user-specific tests in Report Portal
     """
+    rp_url = settings.report_portal.portal_url or config.getini('rp_endpoint')
+    rp_uuid = getenv('RP_UUID') or config.getini('rp_uuid') or settings.report_portal.api_key
+    rp_launch_name = (
+        getenv('RP_LAUNCH') or settings.report_portal.launch_name or config.getini('rp_launch')
+    )
+    rp_project = config.getini('rp_project') or settings.report_portal.project
     fail_args = config.getoption('only_failed', False)
     skip_arg = config.getoption('only_skipped', False)
     user_arg = config.getoption('user', False)
-    upgrades_rerun = config.getoption('upgrades_rerun', False)
-    if not any([fail_args, skip_arg, user_arg, upgrades_rerun]):
+    sat_version = config.getoption('sat_version', None)
+    ref_launch_uuid = config.getoption('rp_reference_launch_uuid', None) or config.getoption(
+        'rp_rerun_of', None
+    )
+    tests = []
+    if not any([fail_args, skip_arg, user_arg]):
         return
-    rp = ReportPortal()
-    version = settings.server.version
-    sat_version = f'{version.base_version}.{version.epoch}'
-    logger.info(f'Fetching Report Portal launches for target Satellite version: {sat_version}')
-    launch = next(
-        iter(
-            rp.launches(
-                sat_version=sat_version, launch_type='upgrades' if upgrades_rerun else 'satellite6'
-            ).values()
+    rp = ReportPortal(rp_url=rp_url, rp_uuid=rp_uuid, rp_project=rp_project)
+
+    if ref_launch_uuid:
+        logger.info(f'Fetching A reference Report Portal launch {ref_launch_uuid}')
+        try:
+            ref_launches = rp.get_launches(uuid=ref_launch_uuid)
+        except IndexError:
+            raise pytest.UsageError(
+                f'Provided reference launch {ref_launch_uuid} was not found or is not finished'
+            )
+    else:
+        if not sat_version:
+            sat_release = getenv('SAT_VERSION') or settings.server.version.get('release', '')
+            sat_snap = getenv('SAT_SNAP') or settings.server.version.get('snap', '')
+            if not all([sat_release, sat_snap]):
+                raise pytest.UsageError(
+                    '--failed|skipped-only requires a reference launch id or'
+                    ' satellite version to be provided'
+                )
+            sat_version = f'{sat_release}-{sat_snap}'
+        logger.info(
+            f'Fetching A reference Report Portal launch by Satellite' f' version: {sat_version}'
```

Comment:
I think this should be moved above all along with `ref_launch_uuid` and this whole code can be refactored to fail fast, like:

```suggestion
            if not (ref_launch_uuid or all([sat_release, sat_snap])):
                raise pytest.UsageError(
                    '--failed|skipped-only requires a reference launch id or'
                    ' satellite version to be provided'
                )
            elif ref_launch_uuid:
                # same code that u have now if ref_launch_uuid is provided
            else:
                # the code just if sat_release and sat_snap provided.
                
```